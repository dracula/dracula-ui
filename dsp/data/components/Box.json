{
  "class": "component",
  "type": "page",
  "id": "Box",
  "name": "Box",
  "last_updated": "2021-03-14T00:08:30.329Z",
  "last_updated_by": "System",
  "description": "Box is the most primitive component of Dracula UI.\nUsing Box allows for consumers of the components library to compose\nmore complex patterns, components, and UIs.\n\nBox includes built-in Color and Spacing properties that make building\ncomplex components convenient and consistent.",
  "related_entity_ids": [],
  "tags": ["component"],
  "snippets": {
    "trigger": "box",
    "languages": {
      "javascriptreact": "<Box color=\"purple\">Dracula</Box>;\n",
      "html": "<div color=\"purple\" class=\"drac-box drac-bg-purple\">Dracula</div>\n"
    }
  },
  "ext_com_draculaui_variations": [
    {
      "title": "Usage",
      "html": "<div color=\"purple\" class=\"drac-box drac-bg-purple\">Dracula</div>\n",
      "react": "<Box color=\"purple\">Dracula</Box>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Box.png",
      "svgData": "./dsp/assets/svgs/Box.svg"
    },
    {
      "title": "Colors",
      "html": "<div color=\"pinkPurple\" class=\"drac-box drac-bg-pink-purple\">Pink Purple</div>\n",
      "react": "<Box color=\"pinkPurple\">Pink Purple</Box>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxColors.png",
      "svgData": "./dsp/assets/svgs/BoxColors.svg"
    },
    {
      "title": "Rounded",
      "html": "<div style=\"display: flex; flex-direction: row\">\n  <div\n    color=\"purple\"\n    rounded=\"none\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-none\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"sm\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-sm\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"base\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"lg\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-lg\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"2xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-2xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"3xl\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-3xl\"\n  ></div>\n  <div\n    color=\"purple\"\n    rounded=\"full\"\n    style=\"width: 250px; height: 250px; margin-right: 20px\"\n    class=\"drac-box drac-bg-purple drac-rounded-full\"\n  ></div>\n</div>\n",
      "react": "<div style={{ display: \"flex\", flexDirection: \"row\" }}>\n  <Box\n    color=\"purple\"\n    rounded=\"none\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"sm\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"base\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"lg\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"2xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"3xl\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n  <Box\n    color=\"purple\"\n    rounded=\"full\"\n    style={{ width: 250, height: 250, marginRight: 20 }}\n  />\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxRounded.png",
      "svgData": "./dsp/assets/svgs/BoxRounded.svg"
    },
    {
      "title": "Generic Elements",
      "html": "<div>\n  <button as=\"button\" color=\"pinkPurple\" class=\"drac-box drac-bg-pink-purple\">\n    Button</button\n  ><a as=\"a\" color=\"green\" href=\"#\" class=\"drac-box drac-bg-green\">Link</a>\n</div>\n",
      "react": "<div>\n  <Box as=\"button\" color=\"pinkPurple\">\n    Button\n  </Box>\n  <Box as=\"a\" color=\"green\" href=\"#\">\n    Link\n  </Box>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/BoxGeneric Elements.png",
      "svgData": "./dsp/assets/svgs/BoxGeneric Elements.svg"
    }
  ],
  "ext_com_draculaui_props": {
    "color": {
      "defaultValue": null,
      "description": "The background color.",
      "name": "color",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": {
        "name": "\"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
      }
    },
    "rounded": {
      "defaultValue": null,
      "description": "The border radius.",
      "name": "rounded",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"sm\" | \"base\" | \"lg\" | \"xl\" | \"2xl\" | \"3xl\" | \"full\" | undefined"
      }
    },
    "spacing": {
      "defaultValue": null,
      "description": "Dracula UI standard spacing properties.",
      "name": "spacing",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "SpacingPropType | undefined" }
    },
    "margin": {
      "defaultValue": null,
      "description": "Dracula UI standard margin properties.",
      "name": "margin",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "MarginPropType | undefined" }
    },
    "as": {
      "defaultValue": null,
      "description": "The HTML element to be used",
      "name": "as",
      "parent": {
        "fileName": "src/components/Box/Box.tsx",
        "name": "BoxProps"
      },
      "declarations": [
        { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
      ],
      "required": false,
      "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
    }
  },
  "ext_com_draculaui_docgen": {
    "tags": {},
    "description": "Box is the most primitive component of Dracula UI.\nUsing Box allows for consumers of the components library to compose\nmore complex patterns, components, and UIs.\n\nBox includes built-in Color and Spacing properties that make building\ncomplex components convenient and consistent.",
    "displayName": "Box",
    "methods": [],
    "props": {
      "color": {
        "defaultValue": null,
        "description": "The background color.",
        "name": "color",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": {
          "name": "\"white\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | undefined"
        }
      },
      "rounded": {
        "defaultValue": null,
        "description": "The border radius.",
        "name": "rounded",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"sm\" | \"base\" | \"lg\" | \"xl\" | \"2xl\" | \"3xl\" | \"full\" | undefined"
        }
      },
      "spacing": {
        "defaultValue": null,
        "description": "Dracula UI standard spacing properties.",
        "name": "spacing",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "SpacingPropType | undefined" }
      },
      "margin": {
        "defaultValue": null,
        "description": "Dracula UI standard margin properties.",
        "name": "margin",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "MarginPropType | undefined" }
      },
      "as": {
        "defaultValue": null,
        "description": "The HTML element to be used",
        "name": "as",
        "parent": {
          "fileName": "src/components/Box/Box.tsx",
          "name": "BoxProps"
        },
        "declarations": [
          { "fileName": "src/components/Box/Box.tsx", "name": "BoxProps" }
        ],
        "required": false,
        "type": { "name": "keyof HTMLElementTagNameMap | undefined" }
      }
    }
  }
}
