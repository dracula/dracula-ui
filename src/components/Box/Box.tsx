import cx from 'classnames/dedupe'
import React, { AllHTMLAttributes } from 'react'
import { borderColors, colors, glowColors } from '../../base/colors'
import {
  PaddingMixin,
  paddingMixin,
  marginMixin,
  MarginMixin
} from '../../base/spacing'

type Element = HTMLElementTagNameMap

export const roundedBorders = {
  none: 'drac-rounded-none',
  sm: 'drac-rounded-sm',
  base: 'drac-rounded',
  lg: 'drac-rounded-lg',
  xl: 'drac-rounded-xl',
  '2xl': 'drac-rounded-2xl',
  '3xl': 'drac-rounded-3xl',
  full: 'drac-rounded-full'
}

/**
 * Box Props
 */
export type BoxProps<K extends keyof Element = 'div'> = {
  /** The background color. */
  color?: keyof typeof colors

  /** The glow color. */
  glowColor?: keyof typeof glowColors

  /** The border color. */
  borderColor?: keyof typeof borderColors

  /** The border radius. */
  rounded?: keyof typeof roundedBorders

  /** The HTML element to be used */
  as?: K
} & AllHTMLAttributes<K> &
  PaddingMixin &
  MarginMixin

/**
 * Box is the most primitive component of Dracula UI.
 * Using Box allows for consumers of the components library to compose
 * more complex patterns, components, and UIs.
 *
 * Box includes built-in Color and Spacing properties that make building
 * complex components convenient and consistent.
 */
export function Box<T extends keyof Element>(props: BoxProps<T>) {
  const finalProps = {
    ...props,
    className: cx(
      `drac-box`,
      props.className,
      props.color && colors[props.color],
      props.glowColor && glowColors[props.glowColor],
      props.borderColor && borderColors[props.borderColor],
      props.rounded && roundedBorders[props.rounded],
      ...paddingMixin(props),
      ...marginMixin(props)
    )
  }

  const as = finalProps.as ?? 'div'
  return React.createElement(as, finalProps, props.children)
}

Box.displayName = 'Box'
