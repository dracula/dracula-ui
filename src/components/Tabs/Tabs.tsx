import cx from 'classnames/dedupe'
import React, { HTMLAttributes } from 'react'
import { ColorMap } from '../../base/colors'
import {
  cleanProps,
  MarginMixin,
  marginMixin,
  PaddingMixin,
  paddingMixin
} from '../../base/spacing'

export const tabsColors: Partial<ColorMap> = {
  white: 'drac-tabs-white',
  cyan: 'drac-tabs-cyan',
  green: 'drac-tabs-green',
  orange: 'drac-tabs-orange',
  pink: 'drac-tabs-pink',
  purple: 'drac-tabs-purple',
  red: 'drac-tabs-red',
  yellow: 'drac-tabs-yellow'
}

/**
 * Tabs Props
 */
export interface TabsProps
  extends HTMLAttributes<HTMLUListElement>,
    PaddingMixin,
    MarginMixin {
  /** The Dracula UI color for the Tabs. */
  color?: keyof typeof tabsColors
}

/**
 * Tabs are horizontal navigation elements used to display
 * content within the same page.
 */
export const Tabs: React.FC<TabsProps> = (props: TabsProps) => {
  const { color, ...htmlProps } = props
  const finalProps = {
    ...htmlProps,
    className: cx(
      'drac-tabs',
      props.className,
      tabsColors[color ?? 'white'],
      ...paddingMixin(props),
      ...marginMixin(props)
    )
  }

  return <ul {...cleanProps(finalProps)} />
}

Tabs.displayName = 'Tabs'
