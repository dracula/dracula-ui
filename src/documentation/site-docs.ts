import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from './render-component'
import React from 'react'
import fs from 'fs-extra'
import { pretty } from './pretty'
import { getDocGen } from '../documentation/docgen/doc-generator'
import { toDSP } from '../documentation/dsp/component-generator'
import puppeteer from 'puppeteer'
import { componentScreenshot } from './screenshots/component-screenshot'
import os from 'os'

interface Documentation {
  basic: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder> | SnapshotBuilder
}

jest.setTimeout(10000)

const RECORDING = process.env.RECORDING

export function docs<T>(
  componentClass: React.FunctionComponent<T> | string,
  documentation: Documentation
) {
  const name =
    typeof componentClass === 'string'
      ? componentClass
      : componentClass.displayName ?? componentClass.name

  const dspPath = `${process.cwd()}/dsp`

  describe(`Site: ${name}`, () => {
    let browser: puppeteer.Browser

    beforeAll(async () => {
      if (!RECORDING) {
        return
      }

      let executablePath = undefined
      if (os.arch() === 'arm64') {
        executablePath = '/Applications/Chromium.app/Contents/MacOS/Chromium'
      }

      browser = await puppeteer.launch({
        headless: true,
        devtools: true,
        executablePath
      })
    })

    let examples: Record<string, ComponentExample> = {}

    afterAll(async () => {
      if (!RECORDING) {
        return
      }

      await browser.close()
      const docGen = getDocGen(name)
      const dsp = toDSP(name, examples, docGen)

      await fs.mkdirp(`${dspPath}/data/components/`).catch(() => {
        // folder already exists
      })

      const path = `${dspPath}/data/components/${name}.json`
      await fs.writeFile(path, pretty(JSON.stringify(dsp), 'json'))
    })

    test('Basic Usage', async () => {
      const variation = documentation.basic()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()

      if (RECORDING) {
        const [screenshot, svg] = await componentScreenshot(
          browser,
          snapshot,
          null,
          name
        )

        snapshot.screenshot = screenshot
        snapshot.svgData = svg
      }

      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations: SnapshotBuilder[] = ([] as SnapshotBuilder[]).concat(
      documentation.variations()
    )

    variations.forEach((variation) => {
      test(variation.title, async () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()

        if (RECORDING) {
          const [screenshot, svg] = await componentScreenshot(
            browser,
            snapshot,
            variation,
            name
          )

          snapshot.screenshot = screenshot
          snapshot.svgData = svg
        }

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
