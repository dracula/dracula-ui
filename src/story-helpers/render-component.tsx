import React from 'react'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { pretty } from './pretty'

Enzyme.configure({ adapter: new Adapter() })

export type ComponentExample = {
  html: string
  docs: string
  title: string
  react?: any
  screenshot?: string
  svgData?: string
}

export type SnapshotBuilder = {
  comp: () => React.ReactElement
  docs: string
  title: string
}

export function buildSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): SnapshotBuilder {
  return {
    title,
    comp,
    docs: docs.trim()
  }
}

export function renderSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): ComponentExample {
  let component = comp()
  const html = mount(component).html()
  let wrapped = false

  // wrap the component in a div when it's not wrapped already
  // we do this so we can get the actual component example to not be shallow rendered.
  // Shallow rendering a parent "div" gives us a nice JSX representation of the example.
  if (component.type !== 'div') {
    component = <div>{component}</div>
    wrapped = true
  }

  let react = shallow(component).debug({ verbose: true })

  if (wrapped) {
    // replace wrapper div ðŸ˜¬
    react = react
      .replace(/^<div>\n/, '')
      .replace(/\n<\/div>$/, '')
      .trim()
  }

  return {
    title,
    html: pretty(html, 'html'),
    react: pretty(react, 'typescript'),
    docs
  }
}
