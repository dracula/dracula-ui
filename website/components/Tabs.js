import { Component } from "react"
import styles from "./Tabs.module.css"
import CodeHighlight from "./CodeHighlight"
import { Box, Tabs } from "@dracula/dracula-ui"

class DocsTabs extends Component {
  constructor(props) {
    super(props)

    this.state = {
      options: [
        { label: "Preview", panel: this.props.section.html },
        { label: "HTML", panel: this.props.section.html },
        { label: "React", panel: this.props.section.react }
      ],
      selectedTab: 0
    }
  }

  onChangeSelectedTab(index) {
    this.setState({ selectedTab: index })
  }

  renderLabels() {
    return this.state.options.map((option, index) => {
      return (
        <li
          key={index}
          className={
            this.state.selectedTab === index ? styles.tabActive : styles.tab
          }
        >
          <a
            className="drac-tab-link drac-text"
            onClick={() => this.onChangeSelectedTab(index)}
          >
            {option.label}
          </a>
        </li>
      )
    })
  }

  renderPanels() {
    return this.state.options.map((option, index) => {
      return (
        <div
          key={index}
          className={
            this.state.selectedTab === index
              ? styles.tabPanelActive
              : styles.tabPanel
          }
        >
          {this.renderPanel(index, option)}
        </div>
      )
    })
  }

  renderPanel(index, option) {
    if (index === 0) {
      return (
        <Box
          className={styles.usage}
          dangerouslySetInnerHTML={{ __html: option.panel }}
        />
      )
    } else if (index === 1) {
      return <CodeHighlight code={option.panel.trim()} language={"html"} />
    } else if (index === 2) {
      return <CodeHighlight code={option.panel.trim()} language={"jsx"} />
    }
  }

  render() {
    return (
      <div>
        <Tabs color="cyan">
          {this.renderLabels()}
        </Tabs>
        <Box className={styles.codeContainer}>
          {this.renderPanels()}
        </Box>
      </div>
    )
  }
}

export default DocsTabs
