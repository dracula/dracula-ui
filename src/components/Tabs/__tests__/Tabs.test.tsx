import React from 'react'
import { docs } from '@/documentation/site-docs'
import { snapshot } from '@/documentation/render-component'
import { Box } from '@/components/Box/Box'
import { Tabs, TabsProps, tabsColors } from '@/components/Tabs/Tabs'

docs(Tabs, {
  basic() {
    return snapshot(
      'Usage',
      Usage,
      'Display different content based on the Tabs behavior.'
    )
  },
  variations: () => snapshot(
    'Colors',
    Colors,
    'Tabs can be customized to use any of the Dracula UI theme colors.'
  )
})

function Usage() {
  return (
    <Tabs>
      {TabsContent()}
    </Tabs>
  )
}

function Colors() {
  return (
    <Box>
      {Object.keys(tabsColors).map((color) => {
        return (
          <Box key={color}>
            <Tabs color={color as TabsProps['color']}>
              {TabsContent()}
            </Tabs>
          </Box>
        )
      })}
    </Box>
  )
}

function TabsContent() {
  return (
    <>
      <li className="drac-tab drac-tab-active">
        <a className="drac-tab-link drac-text" href="#">
          Item One
        </a>
      </li>
      <li className="drac-tab">
        <a className="drac-tab-link drac-text" href="#">
          Item Two
        </a>
      </li>
      <li className="drac-tab">
        <a className="drac-tab-link drac-text" href="#">
          Item Three
        </a>
      </li>
    </>
  )
}