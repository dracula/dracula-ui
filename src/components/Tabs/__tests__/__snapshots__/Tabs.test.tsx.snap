// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Tabs Basic Usage 1`] = `
Object {
  "docs": "Display different content based on the Tabs behavior.",
  "html": "<ul class=\\"drac-tabs drac-tabs-white\\">
  <li class=\\"drac-tab drac-tab-active\\">
    <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item One</a>
  </li>
  <li class=\\"drac-tab\\">
    <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Two</a>
  </li>
  <li class=\\"drac-tab\\">
    <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Three</a>
  </li>
</ul>
",
  "react": "<Tabs>
  <li className=\\"drac-tab drac-tab-active\\">
    <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
      Item One
    </a>
  </li>
  <li className=\\"drac-tab\\">
    <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
      Item Two
    </a>
  </li>
  <li className=\\"drac-tab\\">
    <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
      Item Three
    </a>
  </li>
</Tabs>;
",
  "title": "Usage",
}
`;

exports[`Site: Tabs Colors 1`] = `
Object {
  "docs": "Tabs can be customized to use any of the Dracula UI theme colors.",
  "html": "<div class=\\"drac-box\\">
  <div class=\\"drac-box\\">
    <ul class=\\"drac-tabs drac-tabs-white\\">
      <li class=\\"drac-tab drac-tab-active\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item One</a>
      </li>
      <li class=\\"drac-tab\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Two</a>
      </li>
      <li class=\\"drac-tab\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Three</a>
      </li>
    </ul>
  </div>
  <div class=\\"drac-box\\">
    <ul class=\\"drac-tabs drac-tabs-cyan\\">
      <li class=\\"drac-tab drac-tab-active\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item One</a>
      </li>
      <li class=\\"drac-tab\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Two</a>
      </li>
      <li class=\\"drac-tab\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Three</a>
      </li>
    </ul>
  </div>
  <div class=\\"drac-box\\">
    <ul class=\\"drac-tabs drac-tabs-green\\">
      <li class=\\"drac-tab drac-tab-active\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item One</a>
      </li>
      <li class=\\"drac-tab\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Two</a>
      </li>
      <li class=\\"drac-tab\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Three</a>
      </li>
    </ul>
  </div>
  <div class=\\"drac-box\\">
    <ul class=\\"drac-tabs drac-tabs-orange\\">
      <li class=\\"drac-tab drac-tab-active\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item One</a>
      </li>
      <li class=\\"drac-tab\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Two</a>
      </li>
      <li class=\\"drac-tab\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Three</a>
      </li>
    </ul>
  </div>
  <div class=\\"drac-box\\">
    <ul class=\\"drac-tabs drac-tabs-pink\\">
      <li class=\\"drac-tab drac-tab-active\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item One</a>
      </li>
      <li class=\\"drac-tab\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Two</a>
      </li>
      <li class=\\"drac-tab\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Three</a>
      </li>
    </ul>
  </div>
  <div class=\\"drac-box\\">
    <ul class=\\"drac-tabs drac-tabs-purple\\">
      <li class=\\"drac-tab drac-tab-active\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item One</a>
      </li>
      <li class=\\"drac-tab\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Two</a>
      </li>
      <li class=\\"drac-tab\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Three</a>
      </li>
    </ul>
  </div>
  <div class=\\"drac-box\\">
    <ul class=\\"drac-tabs drac-tabs-red\\">
      <li class=\\"drac-tab drac-tab-active\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item One</a>
      </li>
      <li class=\\"drac-tab\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Two</a>
      </li>
      <li class=\\"drac-tab\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Three</a>
      </li>
    </ul>
  </div>
  <div class=\\"drac-box\\">
    <ul class=\\"drac-tabs drac-tabs-yellow\\">
      <li class=\\"drac-tab drac-tab-active\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item One</a>
      </li>
      <li class=\\"drac-tab\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Two</a>
      </li>
      <li class=\\"drac-tab\\">
        <a class=\\"drac-tab-link drac-text\\" href=\\"#\\">Item Three</a>
      </li>
    </ul>
  </div>
</div>
",
  "react": "<Box>
  <Box>
    <Tabs color=\\"white\\">
      <li className=\\"drac-tab drac-tab-active\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item One
        </a>
      </li>
      <li className=\\"drac-tab\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item Two
        </a>
      </li>
      <li className=\\"drac-tab\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item Three
        </a>
      </li>
    </Tabs>
  </Box>
  <Box>
    <Tabs color=\\"cyan\\">
      <li className=\\"drac-tab drac-tab-active\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item One
        </a>
      </li>
      <li className=\\"drac-tab\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item Two
        </a>
      </li>
      <li className=\\"drac-tab\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item Three
        </a>
      </li>
    </Tabs>
  </Box>
  <Box>
    <Tabs color=\\"green\\">
      <li className=\\"drac-tab drac-tab-active\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item One
        </a>
      </li>
      <li className=\\"drac-tab\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item Two
        </a>
      </li>
      <li className=\\"drac-tab\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item Three
        </a>
      </li>
    </Tabs>
  </Box>
  <Box>
    <Tabs color=\\"orange\\">
      <li className=\\"drac-tab drac-tab-active\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item One
        </a>
      </li>
      <li className=\\"drac-tab\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item Two
        </a>
      </li>
      <li className=\\"drac-tab\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item Three
        </a>
      </li>
    </Tabs>
  </Box>
  <Box>
    <Tabs color=\\"pink\\">
      <li className=\\"drac-tab drac-tab-active\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item One
        </a>
      </li>
      <li className=\\"drac-tab\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item Two
        </a>
      </li>
      <li className=\\"drac-tab\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item Three
        </a>
      </li>
    </Tabs>
  </Box>
  <Box>
    <Tabs color=\\"purple\\">
      <li className=\\"drac-tab drac-tab-active\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item One
        </a>
      </li>
      <li className=\\"drac-tab\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item Two
        </a>
      </li>
      <li className=\\"drac-tab\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item Three
        </a>
      </li>
    </Tabs>
  </Box>
  <Box>
    <Tabs color=\\"red\\">
      <li className=\\"drac-tab drac-tab-active\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item One
        </a>
      </li>
      <li className=\\"drac-tab\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item Two
        </a>
      </li>
      <li className=\\"drac-tab\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item Three
        </a>
      </li>
    </Tabs>
  </Box>
  <Box>
    <Tabs color=\\"yellow\\">
      <li className=\\"drac-tab drac-tab-active\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item One
        </a>
      </li>
      <li className=\\"drac-tab\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item Two
        </a>
      </li>
      <li className=\\"drac-tab\\">
        <a className=\\"drac-tab-link drac-text\\" href=\\"#\\">
          Item Three
        </a>
      </li>
    </Tabs>
  </Box>
</Box>;
",
  "title": "Colors",
}
`;
